packages/server/dataloader/azureDevOpsLoaders.ts

```diff
@@ -0,0 +1,425 @@
+  id: string
+  coreRevision: number
+  revision: number
+  timeStamp: string
+}
+
+export const freshAzureDevOpsAuth = (
+  parent: RootDataLoader
+): DataLoader<TeamUserKey, IGetTeamMemberIntegrationAuthQueryResult | null, string> => {
+  return new DataLoader<TeamUserKey, IGetTeamMemberIntegrationAuthQueryResult | null, string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
+          const azureDevOpsAuthToRefresh = (await parent.get('teamMemberIntegrationAuths').load({
+            service: 'azureDevOps',
+            teamId,
+            userId
+          })) as IGetTeamMemberIntegrationAuthQueryResult | null
+          if (azureDevOpsAuthToRefresh === null) {
+            return null
+          }
+          const {
+            accessToken: existingAccessToken,
+            refreshToken,
+            accessTokenSecret,
+            providerId
+          } = azureDevOpsAuthToRefresh
+          if (!refreshToken) {
+            return null
+          }
+
+          const decodedToken = existingAccessToken && (decode(existingAccessToken) as any)
+          const now = new Date()
+          const inAMinute = Math.floor((now.getTime() + 60000) / 1000)
+          if (!decodedToken || decodedToken.exp < inAMinute) {
+            if (!refreshToken || !accessTokenSecret) {
+              return null
+            }
+            const provider = await parent.get('integrationProviders').loadNonNull(providerId)
+
+            const manager = new AzureDevOpsServerManager(
+              azureDevOpsAuthToRefresh,
+              provider as IntegrationProviderAzureDevOps
+            )
+            const oauthRes = await manager.refresh(refreshToken)
+            if (oauthRes instanceof Error) {
+              return null
+            }
+            const {accessToken, refreshToken: newRefreshToken} = oauthRes
+            const updatedRefreshToken =  newRefreshToken || refreshToken
+            const newAzureDevOpsAuth = {
+              ...azureDevOpsAuthToRefresh,
+              accessToken,
+              refreshToken: updatedRefreshToken
+            }
+            await upsertTeamMemberIntegrationAuth(newAzureDevOpsAuth)
+            return newAzureDevOpsAuth
+          }
+          return azureDevOpsAuthToRefresh
+        })
+      )
+      return results.map((result) => (result.status === 'fulfilled' ? result.value : null))
+    },
+    {
+      ...parent.dataLoaderOptions,
+      cacheKeyFn: (key) => `${key.userId}:${key.teamId}`
+    }
+  )
+}
+
+export const azureDevOpsAllWorkItems = (
+  parent: RootDataLoader
+): DataLoader<TeamUserKey, WorkItem[] | undefined, string> => {
+  return new DataLoader<TeamUserKey, WorkItem[] | undefined, string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
+          const returnWorkItems = [] as WorkItem[]
+          const auth = await parent
```

Comment:
-1 Why not use `const auth = await parent.get('freshAzureDevOpsAuth').load({teamId, userId})`?

Reply:
changed